lix/packages/sdk/src/plugin.ts

```diff
@@ -1,11 +1,40 @@
-import type { LixFile } from "./schema.js";
+import type { Change, Conflict, LixFile } from "./schema.js";
+import type { LixReadonly } from "./types.js";
 
 // named lixplugin to avoid conflict with built-in plugin type
 export type LixPlugin<
-	T extends Record<string, Record<string, unknown>> = Record<string, never>,
+	T extends Record<string, Record<string, unknown>> = Record<string, any>,
 > = {
 	key: string;
 	glob: string;
+	// TODO https://github.com/opral/lix-sdk/issues/37
+	// idea:
+	//   1. runtime reflection for lix on the change schema
+	//   2. lix can validate the changes based on the schema
+	// schema: {
+	// 	bundle: Bundle,
+	// 	message: Message,
+	// 	variant: Variant,
+	// },
+	/**
+	 * Reports if a change from the source lix conflicts with a change in the target lix.
+	 *
+	 * - returns `undefined` if no conflict exists
+	 */
+	reportConflicts?: (args: {
+		sourceLix: LixReadonly;
```

Comment:
i would at lest expect an argument with the "only in source" leaf changes which i think is always needed

Reply:
done